src/nbson_encoder.erl

```diff
@@ -82,7 +82,7 @@ encode_value(V) when is_binary(V) ->
     {?STRING_TYPE, <<?INT32(byte_size(V) + 1), ?CSTRING(V)>>};
 encode_value(V) when is_map(V) ->
     {?EMBDOC_TYPE, encode(V)};
-encode_value(V) when is_list(V), is_tuple(hd(V)), size(hd(V)) == 2 ->
+encode_value(V) when is_list(V), is_tuple(hd(V)), is_binary(element(1, hd(V))) ->
```

Comment:
Why not both guards?

`encode_value(V) when is_list(V), is_tuple(hd(V)), is_binary(element(1, hd(V))), size(hd(V)) == 2 ->`

Reply:
is_binary guard is enough. We are using tuples to represent values of special types, so we can tag them, such as {data, ...} or {object_id, ...}. It is unlikely that we are going to define new data types and even more that we are going to use binaries to tag such new types, so a first element binary on a tuple makes it clear that that tuple is a field of an embdoc.